plugins/modules/azure_rm_virtualmachine.py

```diff
@@ -440,14 +440,23 @@
                     - Flag indicating if boot diagnostics are enabled.
                 required: true
                 type: bool
+            type:
+                description:
+                    - Should the storage account be managed by azure or a custom storage account
+                required: false
+                type: str
+                choices:
+                    - managed
             storage_account:
```

Comment:
since `type` and `storage_account` are conflict with each other, can we add a `mutually_exclusive` contraint.

Reply:
I am unsure how to solve this with mutually_exclusive, it only seems to work on top level options

Reply:
https://github.com/ansible-collections/azure/pull/1206/commits/8409f877aba784a42428d320409873632389444d#diff-050afcc769a0920572289d3cf492f6aa974d9acbb0db79e61ad330ce9fab945fR1125

Reply:
I think if this should be solved generally you'd need to modify

https://github.com/ansible/ansible/blob/eb19692f484b2b58ac3697950eaa4e48b8320db5/lib/ansible/module_utils/common/validation.py#L26-L39

or

https://github.com/ansible/ansible/blob/eb19692f484b2b58ac3697950eaa4e48b8320db5/lib/ansible/module_utils/common/validation.py#L70-L100

to transform the parameters into a list of keys and subkeys in the style of (to fit this module) ['client_id', 'tenant', ...., 'boot_diagnostics.enabled', ....] then you could easily match suboptions. But that feels like a major change that could have performance implications in ansible-core :)

Reply:
other alternative is to move all options to the top level, but that's a bigger breaking change, maybe put that on the todo list for 2.0.0 if you chose to go down that route